ir/src/tests/mod.rs

```diff
@@ -379,6 +379,167 @@ fn transition_constraints_using_parens() {
+        aux: [c, d]
+    public_inputs:
+        stack_inputs: [16]
+    random_values:
+        rand: [m, n[4]]
+    integrity_constraints:
+        enf c' = m + n[2] + $rand[1]
+    boundary_constraints:
+        enf c.first = (n[1] - $rand[0]) * 2
+        enf c.last = m";
+
+    let parsed = parse(source).expect("Parsing failed");
+    let result = AirIR::from_source(&parsed);
+    assert!(result.is_ok());
+}
+
+#[test]
+fn error_random_values_out_of_bounds() {
+    let source_fixed_list = "
+    trace_columns:
+        main: [a, b[12]]
+        aux: [c, d]
+    public_inputs:
+        stack_inputs: [16]
+    random_values:
+        rand: [4]
+    integrity_constraints:
+        enf a' = $rand[4] + 1
+    boundary_constraints:
+        enf a.first = $rand[10] * 2
+        enf a.last = 1";
+
+    let parsed = parse(source_fixed_list).expect("Parsing failed");
+    let result = AirIR::from_source(&parsed);
+    assert!(result.is_err());
+
+    let source_ident_vector_sub_vec = "
+    trace_columns:
+        main: [a, b[12]]
+        aux: [c, d]
+    public_inputs:
+        stack_inputs: [16]
+    random_values:
+        rand: [m, n[4]]
+    integrity_constraints:
+        enf a' = m + 1
+    boundary_constraints:
+        enf a.first = n[5] * 2
+        enf a.last = 1";
+
+    let parsed = parse(source_ident_vector_sub_vec).expect("Parsing failed");
+    let result = AirIR::from_source(&parsed);
+    assert!(result.is_err());
+
+    let source_ident_vector_index = "
+    trace_columns:
+        main: [a, b[12]]
+        aux: [c, d]
+    public_inputs:
+        stack_inputs: [16]
+    random_values:
+        rand: [m, n[4]]
+    integrity_constraints:
+        enf a' = m + 1
+    boundary_constraints:
+        enf a.first = $rand[10] * 2
+        enf a.last = 1";
+
+    let parsed = parse(source_ident_vector_index).expect("Parsing failed");
+    let result = AirIR::from_source(&parsed);
+    assert!(result.is_err());
+}
+
+#[test]
+fn err_random_values_without_aux_cols() {
+    let source = "
+    trace_columns:
+        main: [a, b[12]]
+    public_inputs:
+        stack_inputs: [16]
+    random_values:
+        rand: [16]
+    integrity_constraints:
+        enf c' = $rand[3] + 1
+    boundary_constraints:
+        enf c.first = $rand[10] * 2
+        enf c.last = 1";
+
+    let parsed = parse(source).expect("Parsing failed");
+    let result = AirIR::from_source(&parsed);
+    assert!(result.is_err());
+}
+
+#[test]
+fn err_rand_vals_in_boundary_constr_against_main_cols() {
+    let source = "
+    trace_columns:
+        main: [a, b[12]]
+        aux: [c, d]
+    public_inputs:
+        stack_inputs: [16]
+    random_values:
+        rand: [16]
+    integrity_constraints:
+        enf c' = $rand[3] + 1
+    boundary_constraints:
+        enf a.first = $rand[10] * 2
+        enf b[2].last = 1";
+
+    let parsed = parse(source).expect("Parsing failed");
+    let result = AirIR::from_source(&parsed);
+    assert!(result.is_err());
+}
```

Comment:
Since we're not going to add this check until later, you could add the `ignore` attribute to this test for now and a `TODO` that we'll turn it on later (or it's fine to just remove the test and we can write it again later)